In [6]:
from ultralytics import YOLO
import os
import time
from IPython.display import display, Image
import torch

model = YOLO('yolov8m.pt') 

model.train(
    data='Dog Breeds.yolov8\data.yaml',          
    epochs=100,                  
    imgsz=640,                  
    batch=16,                    
    lr0=0.0003,                  
    optimizer='AdamW',           
    weight_decay=0.0005,         
    save_period=10,              
    patience=20,                 
    amp=True                     
)

metrics_val = model.val()  

print(f"Validation Mean Precision (mP): {metrics_val.box.mp:.4f}")
print(f"Validation Mean Recall (mR): {metrics_val.box.mr:.4f}")
print(f"Validation mAP@0.5: {metrics_val.box.map50:.4f}")
print(f"Validation mAP@0.5:0.95: {metrics_val.box.map:.4f}")

metrics_test = model.val(split='test')  

print(f"Test Mean Precision (mP): {metrics_test.box.mp:.4f}")
print(f"Test Mean Recall (mR): {metrics_test.box.mr:.4f}")
print(f"Test mAP@0.5: {metrics_test.box.map50:.4f}")
print(f"Test mAP@0.5:0.95: {metrics_test.box.map:.4f}")

test_image_path = 'Dog Breeds.yolov8/test/images/Chihuahua_17fbe48c_jpg.rf.d2591965e980c1b6f810d0fed83240da.jpg'
start_time = time.time()
results = model.predict(source=test_image_path, conf=0.25)
end_time = time.time()

output_image_path = 'runs/detect/predict/sample.jpg'
display(Image(filename=output_image_path))

fps = 1 / (end_time - start_time)
print(f"Frames Per Second (FPS): {fps:.2f}")

model.export(format='torchscript') 


<>:14: SyntaxWarning: invalid escape sequence '\d'
<>:14: SyntaxWarning: invalid escape sequence '\d'
C:\Users\spiel\AppData\Local\Temp\ipykernel_24676\1459286248.py:14: SyntaxWarning: invalid escape sequence '\d'
  data='Dog Breeds.yolov8\data.yaml',
100%|██████████| 49.7M/49.7M [00:05<00:00, 9.77MB/s]


Ultralytics 8.3.39  Python-3.12.7 torch-2.5.0+cpu CPU (Intel Core(TM) i7-8750H 2.20GHz)
engine\trainer: task=detect, mode=train, model=yolov8m.pt, data=Dog Breeds.yolov8\data.yaml, epochs=100, time=None, patience=20, batch=16, imgsz=640, save=True, save_period=10, cache=False, device=None, workers=8, project=None, name=train4, exist_ok=False, pretrained=True, optimizer=AdamW, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, s

train: Scanning C:\Users\spiel\OneDrive\Dokumente\Pusan\YOLO Project\datasets\train\labels.cache... 555 images, 0 backgrounds, 0 corrupt: 100%|██████████| 555/555 [00:00<?, ?it/s]
val: Scanning C:\Users\spiel\OneDrive\Dokumente\Pusan\YOLO Project\datasets\valid\labels.cache... 159 images, 0 backgrounds, 0 corrupt: 100%|██████████| 159/159 [00:00<?, ?it/s]


Plotting labels to runs\detect\train4\labels.jpg... 
optimizer: AdamW(lr=0.0003, momentum=0.937) with parameter groups 77 weight(decay=0.0), 84 weight(decay=0.0005), 83 bias(decay=0.0)
TensorBoard: model graph visualization added 
Image sizes 640 train, 640 val
Using 0 dataloader workers
Logging results to runs\detect\train4
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100         0G      0.804      2.172      1.366         22        640: 100%|██████████| 35/35 [24:30<00:00, 42.02s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [02:28<00:00, 29.78s/it]

                   all        159        165      0.325      0.166      0.177     0.0808



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100         0G      0.837      1.637      1.352         23        640: 100%|██████████| 35/35 [16:11<00:00, 27.77s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.16s/it]

                   all        159        165      0.418      0.501      0.409      0.234



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100         0G     0.8114       1.56      1.362         25        640: 100%|██████████| 35/35 [15:16<00:00, 26.19s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.04s/it]

                   all        159        165      0.537      0.545      0.547      0.337



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100         0G      0.809      1.502      1.368         24        640: 100%|██████████| 35/35 [15:15<00:00, 26.15s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.10s/it]

                   all        159        165       0.57      0.557      0.569      0.386



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100         0G     0.8883      1.429      1.394         25        640: 100%|██████████| 35/35 [15:33<00:00, 26.66s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:32<00:00, 18.41s/it]

                   all        159        165      0.595      0.758      0.717      0.485



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100         0G     0.8548      1.328      1.379         29        640: 100%|██████████| 35/35 [15:14<00:00, 26.12s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:32<00:00, 18.51s/it]

                   all        159        165      0.638      0.656      0.724      0.465



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100         0G     0.8095       1.28      1.322         22        640: 100%|██████████| 35/35 [19:52<00:00, 34.07s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:50<00:00, 22.01s/it]

                   all        159        165      0.688      0.617      0.686      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100         0G     0.7725      1.211      1.306         28        640: 100%|██████████| 35/35 [22:47<00:00, 39.08s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [02:25<00:00, 29.16s/it]

                   all        159        165      0.683      0.622      0.761      0.525



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100         0G     0.7827      1.172      1.308         26        640: 100%|██████████| 35/35 [22:48<00:00, 39.09s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:51<00:00, 22.33s/it]

                   all        159        165      0.712      0.783      0.809      0.599



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100         0G     0.8083      1.157       1.33         23        640: 100%|██████████| 35/35 [18:31<00:00, 31.77s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:26<00:00, 17.38s/it]

                   all        159        165      0.607       0.79      0.782      0.556



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100         0G     0.8211      1.119      1.344         30        640: 100%|██████████| 35/35 [15:27<00:00, 26.50s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.12s/it]

                   all        159        165       0.79      0.761      0.837      0.593



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100         0G     0.8204      1.145      1.331         24        640: 100%|██████████| 35/35 [15:18<00:00, 26.25s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:31<00:00, 18.21s/it]

                   all        159        165      0.698      0.782      0.793      0.592



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100         0G     0.7747      1.126       1.31         30        640: 100%|██████████| 35/35 [15:24<00:00, 26.41s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.06s/it]

                   all        159        165      0.834      0.771      0.874      0.624



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100         0G     0.7456       1.07      1.281         21        640: 100%|██████████| 35/35 [15:24<00:00, 26.43s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.01s/it]

                   all        159        165      0.731      0.797      0.846       0.61



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100         0G     0.7405      1.031      1.291         28        640: 100%|██████████| 35/35 [15:18<00:00, 26.25s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.06s/it]

                   all        159        165      0.821      0.753      0.845      0.647



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100         0G     0.7337     0.9641      1.273         25        640: 100%|██████████| 35/35 [15:26<00:00, 26.48s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:31<00:00, 18.34s/it]

                   all        159        165      0.765      0.767      0.851      0.635



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100         0G       0.73      1.004      1.253         35        640: 100%|██████████| 35/35 [19:14<00:00, 33.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [02:23<00:00, 28.71s/it]

                   all        159        165      0.853      0.788      0.871      0.659



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100         0G     0.7048     0.9652      1.256         28        640: 100%|██████████| 35/35 [24:00<00:00, 41.15s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.90s/it]

                   all        159        165      0.735      0.786      0.853      0.654



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100         0G     0.6852     0.9598      1.258         27        640: 100%|██████████| 35/35 [15:18<00:00, 26.25s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.07s/it]

                   all        159        165      0.754       0.86       0.89       0.67



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100         0G     0.6755     0.9236      1.219         32        640: 100%|██████████| 35/35 [15:23<00:00, 26.37s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.12s/it]

                   all        159        165       0.83      0.851      0.926       0.72



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100         0G     0.6831     0.8963      1.239         28        640: 100%|██████████| 35/35 [15:13<00:00, 26.10s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.07s/it]

                   all        159        165       0.68       0.83      0.852      0.652



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100         0G     0.6662     0.8896      1.223         20        640: 100%|██████████| 35/35 [15:13<00:00, 26.09s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.04s/it]

                   all        159        165       0.83      0.801       0.89       0.69



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100         0G     0.6483     0.8914      1.217         30        640: 100%|██████████| 35/35 [15:38<00:00, 26.81s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:31<00:00, 18.21s/it]

                   all        159        165      0.871      0.825      0.911       0.69



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100         0G     0.6605     0.8475      1.237         24        640: 100%|██████████| 35/35 [15:22<00:00, 26.36s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.18s/it]

                   all        159        165      0.835      0.807       0.87      0.671



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100         0G     0.5846     0.7842      1.184         33        640: 100%|██████████| 35/35 [15:21<00:00, 26.33s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.06s/it]

                   all        159        165      0.802      0.851      0.879      0.677



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100         0G     0.6505     0.8575      1.221         36        640: 100%|██████████| 35/35 [15:26<00:00, 26.48s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:32<00:00, 18.53s/it]

                   all        159        165      0.855      0.802      0.878      0.692



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100         0G     0.6367     0.8164        1.2         25        640: 100%|██████████| 35/35 [15:29<00:00, 26.57s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:31<00:00, 18.27s/it]

                   all        159        165      0.884       0.84      0.908      0.668



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100         0G     0.6486     0.7827      1.205         25        640: 100%|██████████| 35/35 [15:17<00:00, 26.20s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:31<00:00, 18.29s/it]

                   all        159        165      0.842      0.884        0.9      0.698



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100         0G     0.6025     0.7732      1.171         27        640: 100%|██████████| 35/35 [15:14<00:00, 26.12s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 18.00s/it]

                   all        159        165      0.858      0.856      0.926      0.748



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100         0G     0.6242     0.7777      1.194         27        640: 100%|██████████| 35/35 [15:15<00:00, 26.16s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.11s/it]

                   all        159        165      0.908      0.838      0.926      0.735



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100         0G     0.5678      0.738      1.142         30        640: 100%|██████████| 35/35 [15:12<00:00, 26.08s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.05s/it]

                   all        159        165      0.836      0.838      0.913      0.718



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100         0G      0.599     0.7595       1.19         26        640: 100%|██████████| 35/35 [15:15<00:00, 26.15s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.14s/it]

                   all        159        165      0.816      0.869      0.901      0.725



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100         0G      0.582     0.7459      1.157         24        640: 100%|██████████| 35/35 [15:26<00:00, 26.48s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.13s/it]

                   all        159        165      0.871      0.817      0.877      0.701



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100         0G     0.6012     0.6959      1.207         26        640: 100%|██████████| 35/35 [15:19<00:00, 26.28s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:31<00:00, 18.29s/it]

                   all        159        165      0.851      0.808      0.884      0.703



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100         0G     0.5769     0.6638      1.172         31        640: 100%|██████████| 35/35 [17:03<00:00, 29.26s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [02:19<00:00, 27.99s/it]

                   all        159        165      0.856      0.878      0.915      0.737



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100         0G     0.5795     0.6446      1.148         25        640: 100%|██████████| 35/35 [23:02<00:00, 39.51s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [02:20<00:00, 28.08s/it]

                   all        159        165      0.858      0.869      0.921      0.706



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100         0G     0.6116     0.7128      1.185         23        640: 100%|██████████| 35/35 [17:09<00:00, 29.40s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.13s/it]

                   all        159        165      0.828      0.877      0.927       0.72



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100         0G     0.5376      0.651      1.129         28        640: 100%|██████████| 35/35 [15:12<00:00, 26.06s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.98s/it]

                   all        159        165      0.877      0.885      0.922      0.745



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100         0G     0.5661     0.6667       1.15         29        640: 100%|██████████| 35/35 [15:22<00:00, 26.35s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:31<00:00, 18.24s/it]

                   all        159        165      0.891      0.861       0.92      0.732



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100         0G     0.5408     0.6464      1.137         36        640: 100%|██████████| 35/35 [15:10<00:00, 26.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.17s/it]

                   all        159        165      0.876       0.82      0.905      0.722



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100         0G     0.5561     0.6428       1.15         25        640: 100%|██████████| 35/35 [15:12<00:00, 26.08s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.87s/it]

                   all        159        165      0.899      0.859      0.905       0.74



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100         0G     0.5591     0.6815      1.147         23        640: 100%|██████████| 35/35 [24:10<00:00, 41.44s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [02:45<00:00, 33.13s/it]

                   all        159        165      0.858      0.882      0.925      0.748



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100         0G     0.5513     0.6142      1.139         30        640: 100%|██████████| 35/35 [20:14<00:00, 34.70s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.01s/it]

                   all        159        165      0.855      0.869      0.918      0.744



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100         0G     0.5465     0.6108      1.161         26        640: 100%|██████████| 35/35 [15:17<00:00, 26.20s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:30<00:00, 18.16s/it]

                   all        159        165      0.853      0.863      0.911      0.728



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100         0G     0.5265      0.642      1.136         23        640: 100%|██████████| 35/35 [15:36<00:00, 26.75s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:34<00:00, 18.82s/it]

                   all        159        165      0.827      0.855      0.902      0.746



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100         0G     0.5436     0.6472      1.142         25        640: 100%|██████████| 35/35 [15:28<00:00, 26.51s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.89s/it]

                   all        159        165      0.853      0.866      0.919      0.749



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100         0G     0.5031     0.6103      1.113         25        640: 100%|██████████| 35/35 [15:04<00:00, 25.84s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.80s/it]

                   all        159        165      0.866      0.889      0.935      0.764



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100         0G     0.5557     0.6282      1.153         36        640: 100%|██████████| 35/35 [14:55<00:00, 25.59s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.93s/it]

                   all        159        165      0.827      0.879      0.913      0.733



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100         0G     0.4865     0.5809      1.097         24        640: 100%|██████████| 35/35 [15:02<00:00, 25.79s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.68s/it]

                   all        159        165       0.87      0.878      0.924      0.739



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100         0G      0.528      0.609      1.122         27        640: 100%|██████████| 35/35 [14:57<00:00, 25.64s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.66s/it]

                   all        159        165      0.887      0.871       0.92      0.764



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100         0G     0.4906     0.5803      1.098         21        640: 100%|██████████| 35/35 [14:58<00:00, 25.67s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.84s/it]

                   all        159        165      0.854      0.878       0.92      0.759



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100         0G     0.5389     0.5825      1.128         25        640: 100%|██████████| 35/35 [14:54<00:00, 25.55s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.86s/it]

                   all        159        165      0.844       0.91      0.938      0.768



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100         0G     0.5261     0.5486      1.127         25        640: 100%|██████████| 35/35 [15:00<00:00, 25.73s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.86s/it]

                   all        159        165      0.873      0.869      0.924      0.751



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100         0G     0.4891     0.5356      1.091         34        640: 100%|██████████| 35/35 [14:55<00:00, 25.60s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.68s/it]

                   all        159        165      0.911      0.905      0.928      0.778



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100         0G     0.4915     0.5326      1.107         25        640: 100%|██████████| 35/35 [14:56<00:00, 25.63s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.66s/it]

                   all        159        165      0.889      0.893      0.941      0.782



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100         0G       0.49      0.532      1.104         35        640: 100%|██████████| 35/35 [14:54<00:00, 25.57s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.89s/it]

                   all        159        165      0.894      0.908      0.949      0.799



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100         0G     0.5035     0.5505      1.134         24        640: 100%|██████████| 35/35 [15:00<00:00, 25.73s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.70s/it]

                   all        159        165      0.932      0.842      0.936      0.786



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100         0G     0.4963     0.5313       1.11         30        640: 100%|██████████| 35/35 [14:53<00:00, 25.52s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.69s/it]

                   all        159        165      0.916      0.901      0.944      0.757



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100         0G     0.5006     0.5301      1.111         30        640: 100%|██████████| 35/35 [14:58<00:00, 25.68s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.72s/it]

                   all        159        165       0.89      0.891      0.938       0.77



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100         0G     0.5075     0.5299      1.113         30        640: 100%|██████████| 35/35 [14:58<00:00, 25.67s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.78s/it]

                   all        159        165      0.898      0.847      0.922      0.759



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100         0G     0.4952     0.5408      1.105         25        640: 100%|██████████| 35/35 [15:03<00:00, 25.80s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.71s/it]

                   all        159        165      0.912      0.875      0.943      0.781



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100         0G     0.4788     0.5153      1.108         19        640: 100%|██████████| 35/35 [14:54<00:00, 25.55s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.69s/it]

                   all        159        165      0.894      0.928      0.954      0.805



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100         0G       0.46     0.4986      1.089         33        640: 100%|██████████| 35/35 [14:58<00:00, 25.68s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.71s/it]

                   all        159        165      0.906       0.92      0.952      0.815



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100         0G     0.4445     0.4895      1.072         29        640: 100%|██████████| 35/35 [14:55<00:00, 25.60s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.80s/it]

                   all        159        165       0.91      0.902       0.95      0.788



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100         0G      0.477     0.5012        1.1         29        640: 100%|██████████| 35/35 [14:56<00:00, 25.63s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.64s/it]

                   all        159        165      0.898      0.869       0.92      0.757



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100         0G     0.4369     0.4557      1.065         27        640: 100%|██████████| 35/35 [14:54<00:00, 25.56s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.65s/it]

                   all        159        165      0.918      0.895      0.939      0.805



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100         0G     0.4475     0.4733      1.071         30        640: 100%|██████████| 35/35 [14:57<00:00, 25.63s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.77s/it]

                   all        159        165      0.868      0.881      0.911      0.791



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100         0G      0.462     0.4845      1.065         23        640: 100%|██████████| 35/35 [15:02<00:00, 25.80s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.90s/it]

                   all        159        165      0.826      0.855      0.899      0.774



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100         0G     0.4454     0.4783      1.071         32        640: 100%|██████████| 35/35 [14:58<00:00, 25.67s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:27<00:00, 17.52s/it]

                   all        159        165       0.92      0.897      0.929      0.762



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100         0G     0.4588     0.4751      1.081         28        640: 100%|██████████| 35/35 [14:51<00:00, 25.48s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:27<00:00, 17.56s/it]

                   all        159        165      0.874      0.915      0.927      0.763



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100         0G     0.4291     0.4616      1.064         30        640: 100%|██████████| 35/35 [14:55<00:00, 25.59s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.77s/it]

                   all        159        165      0.894      0.862      0.916      0.768



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100         0G     0.4257     0.4437      1.064         28        640: 100%|██████████| 35/35 [14:54<00:00, 25.55s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.78s/it]

                   all        159        165      0.906      0.887      0.931      0.782



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100         0G     0.4289     0.4695      1.064         26        640: 100%|██████████| 35/35 [14:48<00:00, 25.40s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:27<00:00, 17.55s/it]

                   all        159        165      0.897        0.9      0.942      0.794



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100         0G     0.4321     0.4464      1.058         26        640: 100%|██████████| 35/35 [14:51<00:00, 25.46s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.65s/it]

                   all        159        165      0.904      0.922      0.943      0.796



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100         0G     0.4332     0.4762      1.056         23        640: 100%|██████████| 35/35 [14:52<00:00, 25.51s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:28<00:00, 17.61s/it]

                   all        159        165      0.909      0.919      0.956      0.797



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100         0G      0.421     0.4349      1.063         29        640: 100%|██████████| 35/35 [16:23<00:00, 28.11s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [02:23<00:00, 28.60s/it]

                   all        159        165      0.907      0.918       0.95      0.802



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100         0G     0.4252     0.4439      1.068         30        640: 100%|██████████| 35/35 [23:54<00:00, 40.98s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [02:05<00:00, 25.06s/it]

                   all        159        165      0.913      0.886      0.942      0.791



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100         0G     0.4026     0.4123      1.049         27        640: 100%|██████████| 35/35 [17:59<00:00, 30.85s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:42<00:00, 20.48s/it]

                   all        159        165      0.868      0.906      0.938      0.788



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100         0G     0.4014     0.4019      1.045         26        640: 100%|██████████| 35/35 [17:12<00:00, 29.49s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:41<00:00, 20.39s/it]

                   all        159        165      0.928      0.873      0.944      0.807



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100         0G     0.4043     0.4263      1.041         24        640: 100%|██████████| 35/35 [17:18<00:00, 29.68s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:40<00:00, 20.13s/it]

                   all        159        165      0.931      0.912      0.944      0.792



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100         0G     0.4114     0.4277       1.06         26        640: 100%|██████████| 35/35 [19:07<00:00, 32.78s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [02:07<00:00, 25.47s/it]

                   all        159        165      0.901      0.925      0.942      0.794



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100         0G     0.4019     0.4321      1.055         33        640: 100%|██████████| 35/35 [18:46<00:00, 32.18s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:43<00:00, 20.77s/it]

                   all        159        165      0.896      0.937      0.943      0.793



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100         0G     0.3768     0.3701      1.029         28        640: 100%|██████████| 35/35 [15:46<00:00, 27.03s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:29<00:00, 17.82s/it]

                   all        159        165      0.897      0.909      0.941      0.799
EarlyStopping: Training stopped early as no improvement observed in last 20 epochs. Best results observed at epoch 63, best model saved as best.pt.
To update EarlyStopping(patience=20) pass a new patience value, i.e. `patience=300` or use `patience=0` to disable EarlyStopping.



83 epochs completed in 24.900 hours.
Optimizer stripped from runs\detect\train4\weights\last.pt, 52.0MB
Optimizer stripped from runs\detect\train4\weights\best.pt, 52.0MB

Validating runs\detect\train4\weights\best.pt...
Ultralytics 8.3.39  Python-3.12.7 torch-2.5.0+cpu CPU (Intel Core(TM) i7-8750H 2.20GHz)
Model summary (fused): 218 layers, 25,842,655 parameters, 0 gradients, 78.7 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [01:20<00:00, 16.05s/it]


                   all        159        165      0.906       0.92      0.952      0.815
             Chihuahua         34         35      0.825       0.81      0.905      0.697
                 Corgi         36         38      0.836      0.947       0.91      0.776
                 Husky         32         32      0.932      0.969      0.988       0.89
             Retriever         26         27      0.937          1      0.994      0.917
                Vizsla         31         33          1      0.874      0.965      0.795
Speed: 4.5ms preprocess, 486.8ms inference, 0.0ms loss, 0.5ms postprocess per image
Results saved to runs\detect\train4
Ultralytics 8.3.39  Python-3.12.7 torch-2.5.0+cpu CPU (Intel Core(TM) i7-8750H 2.20GHz)
Model summary (fused): 218 layers, 25,842,655 parameters, 0 gradients, 78.7 GFLOPs


val: Scanning C:\Users\spiel\OneDrive\Dokumente\Pusan\YOLO Project\datasets\valid\labels.cache... 159 images, 0 backgrounds, 0 corrupt: 100%|██████████| 159/159 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 10/10 [01:19<00:00,  7.96s/it]


                   all        159        165      0.906       0.92      0.952      0.815
             Chihuahua         34         35      0.825       0.81      0.905      0.697
                 Corgi         36         38      0.836      0.947       0.91      0.776
                 Husky         32         32      0.932      0.969      0.988       0.89
             Retriever         26         27      0.937          1      0.994      0.917
                Vizsla         31         33          1      0.874      0.965      0.795
Speed: 4.0ms preprocess, 481.3ms inference, 0.0ms loss, 0.5ms postprocess per image
Results saved to runs\detect\train42
Validation Mean Precision (mP): 0.9059
Validation Mean Recall (mR): 0.9199
Validation mAP@0.5: 0.9522
Validation mAP@0.5:0.95: 0.8151
Ultralytics 8.3.39  Python-3.12.7 torch-2.5.0+cpu CPU (Intel Core(TM) i7-8750H 2.20GHz)


val: Scanning C:\Users\spiel\OneDrive\Dokumente\Pusan\YOLO Project\datasets\test\labels.cache... 79 images, 0 backgrounds, 0 corrupt: 100%|██████████| 79/79 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:38<00:00,  7.62s/it]


                   all         79         82      0.965      0.941      0.973       0.85
             Chihuahua         12         13       0.95      0.846      0.915      0.716
                 Corgi         19         20          1      0.988      0.995      0.875
                 Husky         11         11      0.981      0.909      0.988      0.866
             Retriever         21         21      0.951          1      0.993      0.899
                Vizsla         16         17      0.942      0.961      0.976      0.897
Speed: 3.9ms preprocess, 454.7ms inference, 0.0ms loss, 0.6ms postprocess per image
Results saved to runs\detect\train43
Test Mean Precision (mP): 0.9649
Test Mean Recall (mR): 0.9409
Test mAP@0.5: 0.9733
Test mAP@0.5:0.95: 0.8505

image 1/1 c:\Users\spiel\OneDrive\Dokumente\Pusan\YOLO Project\Dog Breeds.yolov8\test\images\Chihuahua_17fbe48c_jpg.rf.d2591965e980c1b6f810d0fed83240da.jpg: 640x640 1 Chihuahua, 474.7ms
Speed: 3.2ms preprocess, 474.7ms inference, 4.5m

C:\Users\spiel\AppData\Local\Temp\ipykernel_24676\1459286248.py:14: SyntaxWarning: invalid escape sequence '\d'
  data='Dog Breeds.yolov8\data.yaml',


FileNotFoundError: [Errno 2] No such file or directory: 'runs/detect/predict/sample.jpg'

I started with this and got horrible results.

model = YOLO('yolov8s.yaml')  # or 'yolov8n.yaml' for smaller model if memory is a concern

model.train(
    data='/kaggle/input/dog-breed-classification-yolov8/data.yaml',           # Path to your dataset YAML file
    epochs=100,                 # Number of epochs
    imgsz=640,                  # Image size
    batch=32,                   # Batch size, adjust based on GPU capacity
    lr0=0.01,                   # Initial learning rate
    optimizer='AdamW',          # Optimizer
    weight_decay=0.0005,        # Weight decay
    freeze=10,                  # Freeze first 10 layers
    save_period=5,              # Save weights every 5 epochs
    patience=10,                # Early stopping after 10 epochs of no improvement
    amp=True,                   # Automatic mixed precision (faster training)
    device=device               # Specify device
)

Then I used the best.pt and tried different parameters and let it run for more epochs to see if it would improve.
This also did not improve and stopped after 30 epochs. 
model = YOLO('best.pt')

model.train(
    data='Dog Breeds.yolov8\data.yaml',          
    epochs=100,                 # Number of epochs
    imgsz=640,                  # Image size
    batch=64,                   # Batch size, adjust based on GPU capacity
    lr0=0.001,                   # Initial learning rate
    optimizer='AdamW',          # Optimizer
    weight_decay=0.0005,        # Weight decay
    freeze=20,                  # Freeze first 10 layers
    save_period=5,              # Save weights every 5 epochs
    patience=10,                # Early stopping after 10 epochs of no improvement
    amp=True,                   # Automatic mixed precision (faster training)
)

Results:
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     30/100         0G      2.157      3.364      2.617        106        640: 100%|██████████| 9/9 [03:25<00:00, 22.84s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:44<00:00, 22.36s/it]
                   all        159        165      0.537      0.102     0.0873     0.0251
EarlyStopping: Training stopped early as no improvement observed in last 10 epochs. Best results observed at epoch 20, best model saved as best.pt.
To update EarlyStopping(patience=10) pass a new patience value, i.e. `patience=300` or use `patience=0` to disable EarlyStopping

model.train(
    data='Dog Breeds.yolov8\data.yaml',          
    epochs=100,                 # Number of epochs
    imgsz=640,                  # Image size
    batch=64,                   # Batch size, adjust based on GPU capacity
    lr0=0.001,                   # Initial learning rate
    optimizer='AdamW',          # Optimizer
    weight_decay=0.0005,        # Weight decay
    freeze=20,                  # Freeze first 10 layers
    save_period=5,              # Save weights every 5 epochs
    patience=10,                # Early stopping after 10 epochs of no improvement
    amp=True,                   # Automatic mixed precision (faster training)
)
